src/glum/_distribution.py

```diff
@@ -522,15 +528,35 @@ def _score_matrix(self, link, X, y, mu, sample_weight, dispersion, fit_intercept
         ).reshape(-1, 1)
 
         if fit_intercept:
-            if sparse.issparse(X):
-                return sparse.hstack((W, X.multiply(W)))
-            else:
+            if sparse.sputils.isdense(X):
                 return np.hstack((W, np.multiply(X, W)))
-        else:
-            if sparse.issparse(X):
-                return X.multiply(W)
             else:
+                if isinstance(X, SplitMatrix):
```

Comment:
This is the right idea, but I think it should be implemented in tabmat and not here. What we need here is a simple element-wise multiplication. Some classes already implement the `multiply` method. Let's implement them for the other classes so that here we can simply call `X.multiply(W)`